core/src/main/java/org/mule/runtime/core/processor/chain/AbstractMessageProcessorChain.java

```diff
@@ -121,14 +104,47 @@ protected Event doProcess(Event event) throws MuleException {
 
   @Override
   public Publisher<Event> apply(Publisher<Event> publisher) {
+    List<BiFunction<Processor, Function<Publisher<Event>, Publisher<Event>>, Function<Publisher<Event>, Publisher<Event>>>> interceptorsToBeExecuted =
+        resolveInterceptors();
+
+    Flux<Event> stream = from(publisher);
+    for (Processor processor : getProcessorsToExecute()) {
+      Function<Publisher<Event>, Publisher<Event>> processorFunction = processor;
+      for (BiFunction<Processor, Function<Publisher<Event>, Publisher<Event>>, Function<Publisher<Event>, Publisher<Event>>> interceptor : interceptorsToBeExecuted) {
+        processorFunction = interceptor.apply(processor, processorFunction);
+      }
+      stream = stream.transform(processorFunction);
+    }
+    return stream;
+  }
+
+  private List<BiFunction<Processor, Function<Publisher<Event>, Publisher<Event>>, Function<Publisher<Event>, Publisher<Event>>>> resolveInterceptors() {
+    List<BiFunction<Processor, Function<Publisher<Event>, Publisher<Event>>, Function<Publisher<Event>, Publisher<Event>>>> interceptors =
+        new ArrayList<>();
+
     if (flowConstruct instanceof Pipeline) {
-      interceptors.add(0, (processor, next) -> ((Pipeline) flowConstruct).getProcessingStrategy().onProcessor(processor, next));
+      interceptors.add((processor, next) -> ((Pipeline) flowConstruct).getProcessingStrategy().onProcessor(processor, next));
     }
 
-    List<ProcessorInterceptor> interceptionHandlerChain =
-        muleContext.getProcessorInterceptorManager().getInterceptors();
+    // Handle errors
+    interceptors.add((processor, next) -> stream -> from(stream)
```

Comment:
Does it not make sense to keep these 4 as state and add the flowConstruct one to local state rather than instance state in apply()